### PR TITLE
Refactor SQLGlot workflow, CI/CD, improve pdoc template

### PR DESCRIPTION
**Pull request by GeorgeSittas**
This is an attempt to improve the workflow in SQLGlot.

- Replaced the `run_checks.sh` script with a `Makefile`.
- Added pre-commit hooks for lint checks.
- Moved dev. dependencies in `setup.py`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+.PHONY: install install-dev tests style docs docs-serve
+
+install:
+	pip install sqlglot
+
+install-dev:
+	pip install -e ".[dev]"
+	pre-commit install
```
Comment (tobymao) [inline Makefile:8]: we should not install pre-commit

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+tests:
```
Comment (tobymao) [inline Makefile:10]: make test

File: Makefile (hunk 1 above)
Comment (GeorgeSittas) [reply Makefile:8]: Should we include a separate rule for installing it, or just ignore its installation completely?

Comment (eakmanrq) [reply Makefile:8]: To expand on this: `pre-commit install` actually installs the git hooks for pre-commit which Toby wants to keep optional: https://pre-commit.com/#3-install-the-git-hook-scripts

So just having pre-commit pip-install allows the style command to work for everyone but then having a separate `install-pre-commit:` lets someone opt-in to having the commit hooks run automatically. 
```
install-pre-commit:
	pre-commit install
```

Comment (GeorgeSittas) [reply Makefile:10]: Done.

File: Makefile (hunk 2 above)
Comment (GeorgeSittas) [reply Makefile]: Done.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,22 @@
+.PHONY: install install-dev install-pre-commit test style docs docs-serve
+
+install:
+	pip install sqlglot
```
Comment (barakalon) [inline Makefile:4]: I don't think this is right.

Do we need this?

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+install-pre-commit:
+	pre-commit install
+
+test:
+	python -m unittest
+
+style:
+	pre-commit run --all-files
+
```
Comment (barakalon) [inline Makefile:17]: There should be one command that does both `make test` and `make style`

**Review by barakalon — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (GeorgeSittas) [reply Makefile:4]: I think I messed this one up, probably meant to do something like `pip install -e .`

File: Makefile (hunk 4 above)
Comment (GeorgeSittas) [reply Makefile:17]: What about `make checks`?

Comment (GeorgeSittas) [reply Makefile:17]: Done.

File: README.md [hunk 5]
```diff
@@ -40,31 +40,30 @@ pip3 install sqlglot
...
-pdoc sqlglot --docformat google
+make docs-serve
 ```
 
 ## Run Tests and Lint
 
 ```
-# set `SKIP_INTEGRATION=1` to skip integration tests
-./run_checks.sh
+make style
```
Comment (barakalon) [inline README.md:63]: `make check`

**Review by tobymao — APPROVED** [review]

**Pull request merged**